src/app/components/server/StatusPill.tsx

```diff
@@ -40,26 +38,25 @@ export const StatusPill = (props: Props) => {
   );
 };
 
-function hasDirectType(props: Props): props is DirectTypeProps {
-  return typeof (props as DirectTypeProps).type === "string";
-}
-/* c8 ignore stop */
-
-export const getExposureStatus = (exposure: Exposure): StatusPillType => {
+export const getExposureStatus = (
+  exposure: Exposure,
+  isPremium: boolean,
+): StatusPillType => {
   if (isScanResult(exposure)) {
-    if (exposure.manually_resolved) {
+    if (exposure.manually_resolved || exposure.status === "removed") {
       return "fixed";
     }
 
-    switch (exposure.status) {
-      case "removed":
-        return "fixed";
-      case "waiting_for_verification":
-      case "optout_in_progress":
-        return "progress";
-      default:
-        return "needAction";
+    if (
+      isPremium &&
```

Comment:
I’m not sure if this is the correct fix for the issue.  I understand that Premium users _should_ have all their exposures fixed automatically, but I’d like to understand better why the current exposure does not match any of the statuses below? If the exposure is not removed or scheduled for removal it should not show in the list of fixed issues.